packages/core/admin/admin/src/components/LeftMenu.tsx

```diff
@@ -45,115 +65,56 @@ const LeftMenu = ({ generalSectionLinks, pluginsSectionLinks }: LeftMenuProps) =
-                      aria-label={labelValue}
-                    >
-                      <NavLink.Icon label={labelValue}>
-                        <LinkIcon width="2rem" height="2rem" fill="neutral500" />
-                      </NavLink.Icon>
-                      {badgeContent && (
-                        <NavLink.Badge
-                          label="locked"
-                          background="transparent"
-                          textColor="neutral500"
-                        >
-                          {badgeContent}
-                        </NavLink.Badge>
-                      )}
-                    </NavLink.Link>
-                  </NavLink.Tooltip>
-                </Flex>
-              );
-            })
-          : null}
-        {generalSectionLinks.length > 0
-          ? generalSectionLinks.map((link) => {
+        {listLinks.length > 0
+          ? listLinks.map((link) => {
               const LinkIcon = link.icon;
+              const badgeContentLock = link?.lockIcon ? <Lock /> : undefined;
 
-              const badgeContent =
+              const badgeContentNumeric =
                 link.notificationsCount && link.notificationsCount > 0
                   ? link.notificationsCount.toString()
                   : undefined;
 
               const labelValue = formatMessage(link.intlLabel);
-
               return (
                 <Flex tag="li" key={link.to}>
                   <NavLink.Tooltip label={labelValue}>
                     <NavLink.Link
-                      aria-label={labelValue}
                       to={link.to}
                       onClick={() => handleClickOnLink(link.to)}
+                      aria-label={labelValue}
                     >
                       <NavLink.Icon label={labelValue}>
                         <LinkIcon width="2rem" height="2rem" fill="neutral500" />
                       </NavLink.Icon>
-                      {badgeContent && (
-                        <NewNavLinkBadge label={badgeContent} backgroundColor="primary600">
-                          {badgeContent}
-                        </NewNavLinkBadge>
-                      )}
+                      {badgeContentLock ? (
+                        <NavLinkBadgeLock
+                          label="locked"
+                          textColor="neutral500"
+                          paddingLeft={0}
+                          paddingRight={0}
+                        >
+                          {badgeContentLock}
+                        </NavLinkBadgeLock>
+                      ) : badgeContentNumeric ? (
+                        <NavLinkBadgeCounter
+                          label={badgeContentNumeric}
+                          backgroundColor="primary600"
+                          width="2.3rem"
+                          color="neutral0"
+                        >
+                          {badgeContentNumeric}
+                        </NavLinkBadgeCounter>
+                      ) : null}
```

Comment:
```suggestion
                      {link?.lockIcon ? (
                        <NavLinkBadgeLock
                          label="locked"
                          textColor="neutral500"
                          paddingLeft={0}
                          paddingRight={0}
                        >
                          <Lock />
                        </NavLinkBadgeLock>
                      ) : null}
                      {badgeContentNumeric ? (
                        <NavLinkBadgeCounter
                          label={notificationCount}
                          backgroundColor="primary600"
                          width="2.3rem"
                          color="neutral0"
                        >
                          {notificationCount}
                        </NavLinkBadgeCounter>
                      ) : null}
```

Reply:
if people really want to add them both, let them, easier to read.

Reply:
they occupy the same place, if we let them to add both then we will have some ui issues, and also the lock icon will be removed from the Main nav in future because we don t want to show the Releases purchase page in the main nav instead we are gonna move it to the Settings (I added the lock icon logic in past just for Releases)